symbolic/src/Data/Macaw/Symbolic/Testing.hs

```diff
@@ -449,121 +600,100 @@ simulateAndVerify goalSolver logger bak execFeatures archInfo archVals binfo mmP
-  -- initial contents) and a totally symbolic stack
-
+simulateFunction bak execFeatures archVals halloc iMem binfo dfi = do
   let ?ptrWidth = WI.knownRepr
-  (initMem, mmConf) <-
-    case mmPreset of
-      DefaultMemModel -> do
-        (initMem, memPtrTbl) <-
-          MSM.newMergedGlobalMemoryWith populateRelocation (Proxy @arch) bak
-            endianness MSM.ConcreteMutable mems
-        let mmConf = (MSM.memModelConfig bak memPtrTbl)
-                       { MS.lookupFunctionHandle = lookupFunction archVals binfo
-                       , MS.lookupSyscallHandle = lookupSyscall
-                       , MS.mkGlobalPointerValidityAssertion = validityCheck
-                       }
-        pure (initMem, mmConf)
-      LazyMemModel -> do
-        (initMem, memPtrTbl) <-
-          MSMLazy.newMergedGlobalMemoryWith populateRelocation (Proxy @arch) bak
-            endianness MSMLazy.ConcreteMutable mems
-        let mmConf = (MSMLazy.memModelConfig bak memPtrTbl)
-                       { MS.lookupFunctionHandle = lookupFunction archVals binfo
-                       , MS.lookupSyscallHandle = lookupSyscall
-                       , MS.mkGlobalPointerValidityAssertion = validityCheck
-                       }
-        pure (initMem, mmConf)
-
-  -- Allocate a stack and insert it into memory
-  --
-  -- The stack allocation uses the SMT array backed memory model (rather than
-  -- the Haskell-level memory model)
-  let kib = 1024
-  let mib = 1024 * kib
-  stackSize <- WI.bvLit sym WI.knownRepr (BVS.mkBV WI.knownRepr (2 * mib))
-  (MSS.ArrayStack stackBasePtr _stackTopPtr _stackArrayStorage, mem1) <-
-    MSS.createArrayStack bak initMem (MSS.ExtraStackSlots 0) stackSize
-
-  -- Make initial registers, including setting up a stack pointer.
-  --
-  -- The stack pointer points to the middle of the stack allocation. This is a
-  -- hack. We do this because each architecture has some expected stack layout
-  -- (e.g., x86_64 expects a return address just above the stack pointer;
-  -- PPC expects a "back chain"; and AArch32, PPC, and x86_64 all expect
-  -- stack-spilled arguments above everything else). Setting the stack pointer
-  -- to the middle of the allocation allows reads of fully symbolic data from
-  -- above the stack pointer, and this seems to be good enough to make our tests
-  -- pass.
-  --
-  -- The functions in the test-suite do not (and should not) rely on accessing
-  -- data in their caller's stack frames, even though that wouldn't cause test
-  -- failures with this setup.
-  initialRegs <- MS.macawAssignToCrucM (mkInitialRegVal symArchFns sym) (MS.crucGenRegAssignment symArchFns)
-  stackInitialOffset <- WI.bvLit sym WI.knownRepr (BVS.mkBV WI.knownRepr mib)
-  sp <- CLM.ptrAdd sym WI.knownRepr stackBasePtr stackInitialOffset
-  let initialRegsEntry = CS.RegEntry regsRepr initialRegs
-  let regsWithStack = MS.updateReg archVals initialRegsEntry MC.sp_reg sp
-
+  InitialMem initMem mmConf <- pure iMem
```

Comment:
Using `pure` here just to pattern match on `InitialMem` feels silly. What not just:

```suggestion
  let InitialMem initMem mmConf = iMem
```